rackspace/autoscale/v1/webhooks/results.go

```diff
@@ -0,0 +1,112 @@
+package webhooks
+
+import (
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/pagination"
+)
+
+type webhookResult struct {
+	gophercloud.Result
+}
+
+// Extract interprets any webhookResult as a Webhook, if possible.
+func (r webhookResult) Extract() (*Webhook, error) {
+	if r.Err != nil {
+		return nil, r.Err
+	}
+
+	var response struct {
+		Webhook Webhook `mapstructure:"webhook"`
+	}
+
+	err := mapstructure.Decode(r.Body, &response)
+
+	return &response.Webhook, err
+}
+
+// CreateResult represents the result of a create operation. Multiple webhooks
+// can be created in a single call, so the result should be treated as a typical
+// pagination Page.  ExtractWebhooks() can be used to extract a slice of
+// Webhooks from a single page.
+type CreateResult struct {
+	pagination.SinglePageBase
```

Comment:
> Cool. I think I was following the example of rackspace/lb/v1/nodes here.

Hmm, that's a great point. And it so easily refutes everything I just claimed....
I'm still inclined to lean toward my original suggestion. To me, it's a bit unnatural to have users (know to) page through something other than a `ListResult`.

> So, embed gophercloud.Result in place of pagination.SinglePageBase here, and rename the ExtractWebhooks method to just Extract, but still return a slice?

Yes, exactly that.